packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -116,6 +116,19 @@ class ObjectKey extends LocalKey {
 /// You cannot simultaneously include two widgets in the tree with the same
 /// global key. Attempting to do so will assert at runtime.
 ///
+/// ## Pitfalls
+/// It is not advised to instantiate a GlobalKey inside of a build method. Doing
```

Comment:
maybe rephrase this to: GlobalKey are usually long-lived objects owned for example by a [State] object. They should not be re-created on every build.